apps/typegpu-docs/src/content/examples/rendering/3d-fish/index.ts

```diff
@@ -0,0 +1,414 @@
+import tgpu from 'typegpu';
+import { distance } from './tgsl-helpers';
+import * as d from 'typegpu/data';
+import * as std from 'typegpu/std';
+import * as m from 'wgpu-matrix';
+import * as p from './params';
+import {
+  Camera,
+  computeBindGroupLayout,
+  type ModelData,
+  ModelDataArray,
+  modelVertexLayout,
+  MouseRay,
+  renderBindGroupLayout,
+  renderInstanceLayout,
+} from './schemas';
+import { fragmentShader, vertexShader } from './render';
+import { mainCompute } from './compute';
+import { loadModel } from './load-model';
+
+// setup
+
+const canvas = document.querySelector('canvas') as HTMLCanvasElement;
+const context = canvas.getContext('webgpu') as GPUCanvasContext;
+const presentationFormat = navigator.gpu.getPreferredCanvasFormat();
+const root = await tgpu.init();
+
+context.configure({
+  device: root.device,
+  format: presentationFormat,
+  alphaMode: 'premultiplied',
+});
+
+// models and textures
+
+// https://free3d.com/3d-model/fish---low-poly-82864.html
+const fishModel = await loadModel(
+  root,
+  'assets/3d-fish/fish.obj',
+  'assets/3d-fish/fish.png',
+);
+
+// https://www.cgtrader.com/free-3d-models/space/other/rainy-ocean
+// https://www.istockphoto.com/pl/obrazy/sand
+const oceanFloorModel = await loadModel(
+  root,
+  'assets/3d-fish/ocean_floor.obj',
+  'assets/3d-fish/ocean_floor.jpg',
+);
+
+// buffers
+
+const fishDataBuffers = Array.from({ length: 2 }, () =>
+  root
+    .createBuffer(ModelDataArray(p.fishAmount))
+    .$usage('storage', 'uniform', 'vertex'),
+);
```

Comment:
I think it's never used as a uniform
```suggestion
const fishDataBuffers = Array.from({ length: 2 }, () =>
  root.createBuffer(ModelDataArray(p.fishAmount)).$usage('storage', 'vertex'),
);
```